pkg/operator/baremetal_pod.go

```diff
@@ -46,14 +56,48 @@ func setMariadbPassword() corev1.EnvVar {
 		ValueFrom: &corev1.EnvVarSource{
 			SecretKeyRef: &corev1.SecretKeySelector{
 				LocalObjectReference: corev1.LocalObjectReference{
-					Name: "mariadb-password",
+					Name: baremetalSecretName,
 				},
-				Key: "password",
+				Key: baremetalSecretKey,
 			},
 		},
 	}
 }
 
+func generateRandomPassword() string {
+	rand.Seed(time.Now().UnixNano())
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789")
+	length := 16
+	buf := make([]rune, length)
+	for i := range buf {
+		buf[i] = chars[rand.Intn(len(chars))]
+	}
+	return (string(buf))
+}
+
+func createMariadbPasswordSecret(client coreclientv1.SecretsGetter, config *OperatorConfig) error {
+	glog.V(3).Info("Checking if the Maridb password secret already exists")
+	_, err := client.Secrets(config.TargetNamespace).Get(baremetalSecretName, metav1.GetOptions{})
```

Comment:
I'd rather return early here, if err != nil return

Reply:
"return err" in line # 98 takes care of returning the error when the error is something other than IsNotFound() and also the case when there is no error at all.